File: iMEGA/MEGAChatResultDelegate.m
#import "MEGAChatResultDelegate.h"

@interface MEGAChatResultDelegate ()

@property (nonatomic, copy) void (^completion)(MEGAChatSdk *api, uint64_t chatId , MEGAChatConnection newState);

@end

@implementation MEGAChatResultDelegate

- (instancetype)initWithCompletion:(void (^)(MEGAChatSdk *api, uint64_t chatId , MEGAChatConnection newState))completion {
    self = [super init];
    if (self) {
        _completion = completion;
    }
    
    return self;
}

- (void)onChatConnectionStateUpdate:(MEGAChatSdk *)api chatId:(uint64_t)chatId newState:(MEGAChatConnection)newState {
    if (self.completion) {
        self.completion(api, chatId, newState);
    }
}

@end

File: iMEGA/API/Chat/MEGAChatResultDelegate.swift
import Foundation
import MEGADomain

typealias MEGAChatResultDelegateCompletion = (_ api: MEGAChatSdk, _ chatId: HandleEntity, _ newState: MEGAChatConnection) -> Void

class MEGAChatResultDelegate: NSObject, MEGAChatDelegate {
    let completion: MEGAChatResultDelegateCompletion
    
    init(completion: @escaping MEGAChatResultDelegateCompletion) {
        self.completion = completion
    }
    
    func onChatConnectionStateUpdate(_ api: MEGAChatSdk, chatId: UInt64, newState: Int32) {
        guard let intNewState = MEGAChatConnection(rawValue: Int(newState)) else {
            return
        }
        completion(api, chatId, intNewState)
    }
}
